rs/rosetta-api/icp/tests/system_tests/test_cases/neuron_management.rs

```diff
@@ -816,3 +822,154 @@ fn test_get_neuron_info() {
+                    env.rosetta_client
+                        .remove_hot_key(
+                            env.network_identifier.clone(),
+                            &(*TEST_IDENTITY).clone(),
+                            RosettaHotKeyArgs::builder(neuron_index)
+                                .with_principal_id(hot_key_identity.sender().unwrap().into())
+                                .build(),
+                        )
+                        .await
+                        .unwrap();
+
+                    let neuron = list_neurons(&agent)
+                        .await
+                        .full_neurons
+                        .first()
+                        .unwrap()
+                        .to_owned();
+                    // Make sure that the neuron has no hot keys
+                    assert!(neuron.hot_keys.is_empty());
+                    let hot_key_identity = Arc::new(hot_key_identity);
+                    // Now we try the same but with a public key
+                    env.rosetta_client
+                        .add_hot_key(
+                            env.network_identifier.clone(),
+                            &(*TEST_IDENTITY).clone(),
+                            RosettaHotKeyArgs::builder(neuron_index)
+                                .with_public_key((&hot_key_identity).into())
+                                .build(),
+                        )
+                        .await
+                        .unwrap();
+
+                    let neuron = list_neurons(&agent)
+                        .await
+                        .full_neurons
+                        .first()
+                        .unwrap()
+                        .to_owned();
+                    // Make sure that the neuron has a hot key
+                    assert_eq!(neuron.hot_keys.len(), 1);
+                    assert_eq!(
+                        neuron.hot_keys.first().unwrap().0,
+                        hot_key_identity.sender().unwrap()
+                    );
+
+                    // Remove the hot key from the neuron
+                    env.rosetta_client
+                        .remove_hot_key(
+                            env.network_identifier.clone(),
+                            &(*TEST_IDENTITY).clone(),
+                            RosettaHotKeyArgs::builder(neuron_index)
+                                .with_public_key((&hot_key_identity).into())
+                                .build(),
+                        )
+                        .await
+                        .unwrap();
+
+                    let neuron = list_neurons(&agent)
+                        .await
+                        .full_neurons
+                        .first()
+                        .unwrap()
+                        .to_owned();
+                    // Make sure that the neuron has no hot keys
+                    assert!(neuron.hot_keys.is_empty());
+                });
```

Comment:
Let's also test the behavior for invalid inputs: no key provided and both keys provided.